### PR TITLE
[PVR] Fix recordings url encoding.

### PR DESCRIPTION
**Pull request by ksooo**
Regression introduced by #10438 => 5ba26b9c21c665310d3ce0424958069f29f26b99

If a recording folder name contained "?", the children of this folder were never displayed.

This PR fixes this bug.

Changes were runtime tested on macOS, latest kodi master.

screenshot000
screenshot001

@Jalle19 for code review?
@fetzerch fyi, maybe you can runtime test.

### CODE REVIEW

File: xbmc/pvr/recordings/PVRRecordingsPath.cpp [hunk 1]
```diff
@@ -230,3 +239,17 @@ size_t CPVRRecordingsPath::GetDirectoryPathPosition() const
   }
   // unreachable
 }
+
+void CPVRRecordingsPath::EscapeSegment(std::string &strSegment)
```
Comment (xhaggi) [inline xbmc/pvr/recordings/PVRRecordingsPath.cpp]: Any reason why you don't use `CURL::Decode` and `CURL::Encode`?

**Comment by fetzerch** [comment]
Thanks for the quick fix. Runtime tested here with mythtv, all good :)

File: xbmc/pvr/recordings/PVRRecordingsPath.cpp (hunk 1 above)
Comment (ksooo) [inline xbmc/pvr/recordings/PVRRecordingsPath.cpp]: well, i was wrong, i was misinterpreting what it does when i was first reading the code. will change, thanks for the hint.

Comment (ksooo) [inline xbmc/pvr/recordings/PVRRecordingsPath.cpp]: done

**Comment by ksooo** [comment]
second commit fixes estuary recordings window sidebar for folders with special characters in it's name.

before:

screenshot000

after:

screenshot002

@phil65 fyi

**Comment by ksooo** [comment]
jenkins build this please

File: xbmc/pvr/recordings/PVRRecordingsPath.h [hunk 2]
```diff
@@ -52,8 +52,8 @@ namespace PVR
     bool IsDeleted() const { return !IsActive(); }
     bool IsRadio() const { return m_bRadio; }
     bool IsTV() const { return !IsRadio(); }
-    std::string GetDirectoryPath() const { return m_directoryPath; }
-    std::string GetSubDirectoryPath(const std::string &strPath) const;
+    std::string GetUnescapedDirectoryPath() const;
+    std::string GetUnescapedSubDirectoryPath(const std::string &strPath) const;
```
Comment (ksooo) [inline xbmc/pvr/recordings/PVRRecordingsPath.h]: to make it pretty clear that the returned values are unescaped strings.

**Pull request merged**